### PR TITLE
DOC: update documentation for YearBegin

### LINKED ISSUE #52105: BUG: Is offsets.YearBegin working as expected?
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
pd.to_datetime("2017-09-18") - pd.offsets.YearBegin()
#> Timestamp('2017-01-01 00:00:00') - My expectation was correct as I expected exactly the output of the function

pd.to_datetime("2018-01-01") - pd.offsets.YearBegin()
#> Timestamp('2017-01-01 00:00:00') -  My expectation was wrong as I expected Timestamp('2018-01-01 00:00:00')
```

### Issue Description

Hello,

The documentation for `offsets.YearBegin` states "_DateOffset increments between calendar year begin dates."_
I do not understand if my expectations about the behaviour of `offsets.YearBegin` are correct or not. Please see the code below:

```python
pd.to_datetime("2017-09-18") - pd.offsets.YearBegin()
#> Timestamp('2017-01-01 00:00:00') - My expectation was correct as I expected exactly the output of the function

pd.to_datetime("2018-01-01") - pd.offsets.YearBegin()
#> Timestamp('2017-01-01 00:00:00') -  My expectation was wrong as I expected Timestamp('2018-01-01 00:00:00')
```
Why if I subtract `offsets.YearBegin()` from the first day of the year I get the first day of the previous year? Is this expected? 
If so, I could improve the documentation. If not, I guess this could be a bug.

Thanks in advance.

### Expected Behavior

```python
pd.to_datetime("2018-01-01") - pd.offsets.YearBegin()
#> Timestamp('2018-01-01 00:00:00')
```

### Installed Versions

pandas 1.4.4
Python 3.9.16

### PR DESCRIPTION
**Pull request by natmokval**
- [x] closes #52105

Updated documentation for YearBegin, added new examples.

### CODE REVIEW

**Comment by natmokval** [comment]
Hi @MarcoGorelli, could you please review this pr?

**Review by MarcoGorelli — APPROVED** [review]
thanks @natmokval !

File: pandas/_libs/tslibs/offsets.pyx [hunk 1]
```diff
@@ -2313,12 +2313,28 @@ cdef class YearEnd(YearOffset):
 
 cdef class YearBegin(YearOffset):
     """
-    DateOffset increments between calendar year begin dates.
+    DateOffset of one year at beginning.
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/offsets.pyx:2316]: this matches the wording on `MonthBegin`

wondering if there's a clearer wording, but that can be for a separate PR
